sentry_sdk/profiler.py

```diff
@@ -114,9 +110,22 @@ def _sample(self, _, frame):
 
     def to_json(self):
         """
-        Exports this object to a JSON format compatible with Sentry's profiling visualizer
+        Exports this object to a JSON format compatible with Sentry's profiling visualizer.
+        Returns dictionary which can be serialized to JSON.
         """
-        return json.dumps(self, cls=self.JSONEncoder)
+        thread_id = threading.get_ident()
+        return {
+            'samples': [{
+                'frames': sample.stack,
+                'relative_timestamp_ns': sample.sample_time,
+                'thread_id': thread_id
+            } for sample in self.stack_samples],
+            'frames': [{
```

Comment:
We don't need to have the frame list here since it's already captured in the samples.